src/Processors/IProcessor.cpp

```diff
@@ -40,5 +41,37 @@ std::string IProcessor::statusToName(Status status)
     __builtin_unreachable();
 }
 
+void IProcessor::process(bool trace_processors)
+{
+    if (this->workHook)
+        this->workHook->onEnter();
+
+    OpenTelemetry::SpanHolderPtr span;
+    if (trace_processors)
+    {
+        span = std::make_unique<OpenTelemetry::SpanHolder>(demangle(typeid(*this).name()) + "::work()");
+        span->addAttribute("clickhouse.thread_id", CurrentThread::get().thread_id);
+    }
+
+    try
+    {
+        work();
+
+        /// Make sure the span is closed before the hook
+        span.reset();
```

Comment:
`reset()` is noexcept. Let's move it out of the try/catch (e.g. to l. 70).

Besides that, I guess you want to call `reset()` before calling `onLeave()` (l. 66)?